src/frontend/src/lib/components/tokens/TokenExchangeBalance.svelte

```diff
@@ -14,6 +14,8 @@
 		{formatUSD({ value: usdBalance })}
 	{:else if isNullish(balance)}
 		<span class="animate-pulse">{nullishBalanceMessage ?? '-'}</span>
+	{:else if isNullish(usdBalance) && nonNullish(nullishBalanceMessage)}
+		<span class="animate-pulse">{nullishBalanceMessage}</span>
```

Comment:
Is it correct that it should be animated with a pulse effect?

Reply:
adding another comment here, I see that above you do `{nullishBalanceMessage ?? '-'}` and you don't use `nonNullish(nullishBalanceMessage)`

while here the contrary

but in the end, those are the same situations no? why not merging the check?

```typescript
{:else if isNullish(balance) || isNullish(usdBalance)}
		<span class="animate-pulse">{nullishBalanceMessage ?? '-'}</span>
```

Reply:
@peterpeterparker yes, it should be animated with a pulse effect. We already use this pulse effect in the check above and here we want to display the same value.

@AntonioVentilii you are right, we can combine those two checks. I just did it and pushed it.